src/amber/router/params_parser.cr

```diff
@@ -106,7 +107,7 @@ module Amber::Router
         if filename.is_a?(String) && !filename.empty?
           files[upload.name] = Files::File.new(upload: upload)
```

Comment:
For a separate PR this should be move to the params scope beware that the filename can collide with a parameter found in the body or url. Also `ParamsType` would need to include the union `Files::File`

Reply:
`upload.name` is just an a file input name. So sure it could collide but so could any other input field.